### DISCUSSION TITLE
Draw on top of images

### DISCUSSION THREAD

**Discussion by kescobo** [cetz-package/cetz#540] [Q&A] [answered]
For constructing slides especially, it would be very useful to be able to place an image (eg using `#image("path/to/thing.png")` and then place drawings on top of it.

Something like this is possible with tikz, eg:

```    
\begin{tikzpicture}
        \node[inner sep=0pt,anchor=south west] (img1) at (0,0)
            {\includegraphics[width=0.7\textwidth]{hsiao_5a.png}};
        \filldraw[white,very thick,anchor=south west] (2.6,0) rectangle (9.6,4.7);
        % \draw[step=0.2,anchor=south west,gray,thin] (0,0) grid (9.6,7); % to figure out coordinates
        % \draw[step=1.0,anchor=south west] (0,0) grid (9.4,6);
\end{tikzpicture}
```

I tried naively to add a call to `image()` inside a cetz figure,

```
  #cetz.canvas({
    import cetz.draw: *
    // Your drawing code goes here
    image("assets/decision-tree_1.png")
    // Draw a circle and place its "west" anchor at the origin.
    circle((0,0), anchor: "west")
  }) 
```

But this gives

```
error: cannot join content with array
```

Is there any way to accomplish this?

**Comment by johannes-wolf** [comment, marked as answer]
Wrap your `image` call in a `content(...)` call. CeTZ' `content` is a function to display native Typst content inside a CeTZ canvas.

E.g.
```typst
#cetz.canvas({
  import cetz.draw: *
  // Your drawing code goes here
  content((0,0), image("assets/decision-tree_1.png"))
  // Draw a circle and place its "west" anchor at the origin.
  circle((0,0), anchor: "west")
}) 
```

**Reply by kescobo** [reply]
That works great, thank you! I missed that in the manual since there's only text in the examples, but it makes a lot of sense.

Now I need to figure out how to get polylux to animate changes :thinking:
